scripts/check_if_pr_is_low_risk.py

```diff
@@ -0,0 +1,292 @@
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import argparse
+import json
+import re
+import sys
+
+import python_utils
+from scripts import common
+
+
+GITHUB_API_PR_ENDPOINT = (
+    'https://api.github.com/repos/%s/%s/pulls/%s')
+PR_URL_REGEX = (
+    r'^https://github.com/(?P<owner>\w+)/(?P<repo>\w+)/pull/(?P<num>\w+)/?$')
+UPSTREAM_REMOTE = 'upstream'
+
+
+def parse_pr_url(pr_url):
+    """Extract the owner, repo, and PR number from a PR URL.
+
+    For example, in the URL https://github.com/foobar/oppia/pull/23, the
+    owner is `foobar`, the repo is `oppia`, and the PR number is `23`.
+
+    Args:
+        pr_url: str. URL to the PR on GitHub.
+
+    Returns:
+        str, str, str. 3-tuple of the PR owner, repository, and number.
+    """
+    match = re.match(PR_URL_REGEX, pr_url)
+    if match:
+        return match.group('owner', 'repo', 'num')
+    else:
+        return None
+
+
+def load_diff(base_branch):
+    """Load the diff between the head and base.
+
+    Args:
+        base_branch: str. Base branch of PR.
+
+    Returns:
+        tuple(list(tuple(str, str)), dict(str, list(str)). Tuple of a
+        list of changed files (each a tuple of before, after) and a
+        dictionary mapping from file names to list of diff lines for
+        each file. In the event of a parsing error, a tuple of an empty
+        list and empty dictionary is returned.
+    """
+    diff_name_status = common.run_cmd([
+        'git', 'diff', '--name-status',
+        '{}/{}'.format(UPSTREAM_REMOTE, base_branch),
+    ])
+    diff_files = []
+    for line in diff_name_status.split('\n'):
+        if not line:
+            continue
+        split = line.split()
+        if len(split) < 2 or len(split) > 3:
+            python_utils.PRINT(
+                'Failed to parse diff --name-status line "%s"'
+                % line)
+            return [], {}
+        elif len(split) == 2:
+            diff_files.append((split[1], split[1]))
+        elif len(split) == 3:
+            diff_files.append((split[1], split[2]))
+    file_diffs = {}
+    for file_tuple in diff_files:
+        for filename in file_tuple:
+            if filename in file_diffs:
+                continue
+            file_diff = common.run_cmd([
+                'git', 'diff', '-U0',
+                '{}/{}'.format(UPSTREAM_REMOTE, base_branch),
+                '--', filename,
+            ])
+            file_diff_split = file_diff.rstrip().split('\n')
+            i = 0
+            for line in file_diff_split:
+                i += 1
+                if line.startswith('@@'):
+                    break
+            if i > 6 or i == len(file_diff_split):
+                python_utils.PRINT(
+                    'Skipped too many lines when parsing "%s" diff'
+                    % filename)
+                return [], {}
+            file_diffs[filename] = file_diff_split[i:]
+    return diff_files, file_diffs
```

Comment:
Do we need file diffs for all the files, if not should we only fetch it for the required files?

Reply:
done